File: kotlin/src/_1_two_sum.kt
/*
 * @lc app=leetcode id=1 lang=kotlin
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val map = mutableMapOf<Int, Int>()
        for ((i, n) in nums.withIndex()) {
            map[target - n]?.let { return intArrayOf(it, i) }
            map[n] = i
        }
        return intArrayOf()
    }
}
// @lc code=end

File: typescript/src/_1_two_sum.ts
/*
 * @lc app=leetcode id=1 lang=typescript
 *
 * [1] Two Sum
 */

// @lc code=start
function twoSum(nums: number[], target: number): number[] {
    const map = new Map<number, number>();
    for (let i = 0; i < nums.length; i++) {
        const j = map.get(target - nums[i]!);
        if (j !== undefined) {
            return [j, i];
        }
        map.set(nums[i]!, i);
    }
    return [];
};
// @lc code=end
